packages/common/src/types/index.ts

```diff
@@ -73,26 +57,47 @@ export type Store = {
   name: string;
 };
 
+export interface InvoiceLine extends DomainObject {
+  id: string;
+
+  itemId: string;
+  itemName: string;
+  itemCode: string;
+  itemUnit: string;
+  packSize: number;
+  numberOfPacks: number;
+  costPricePerPack: number;
+  sellPricePerPack: number;
+
+  expiryDate?: string | null;
+
+  batch?: string | null;
+
+  stockLine?: StockLine | null;
+
+  location?: string;
+  comment?: string;
+}
```

Comment:
this now aligns with the `InvoiceLineNode`, where it would arguably be better to use that instead. But, for now, I've just converted this over for now.

Personally, apart from the API fns/layer, I would rather not know gql exists throughout the rest of the app, so I would like to use the generated types for typing the api fns, and then map these into our internal types. 

I think it keeps it cleaner and also means when we need to create 'pseudo' entities, we don't have to create an invoice with `lines: { nodes: [] }` and things like that.. 🤷 

There would also be an additional type, though, which has the `update` fn's etc, attached. So, maybe it's getting all a bit complicated

Reply:
it does get a little complicated, though I think worth it. adding in the `update` and other funcs isn't too much of a drama and once we establish a pattern and naming convention then I think it will be pretty obvious
 
I like the separation of the API types from internal types - and would prefer to have a different shape to the types internally, so would like to keep it. 
just worried that there is some duplication, and whether it might bring in some confusion: where someone in the front end codebase is talking about x and that is actually a y in the API / gql which actually maps to database field z. I think all that is pretty minor.